[SimplifyCFG] Do not run `simplifySwitchOfPowersOfTwo` in early invocations

antoniofrighetto opened the pull request:
It may be desirable not to carry out `simplifySwitchOfPowersOfTwo` transform during early pipeline invocations, so as to let other optimizations occur first.

nikic reviewed (commented):
Can you please add a PhaseOrdering test that demonstrates why this is useful?

Also, can we reasonably bind this to one of the existing switch options (like ConvertSwitchToLookupTable)? We have sooo many options already.

nikic commented:
> > Can you please add a PhaseOrdering test that demonstrates why this is useful?
> 
> To simply confirm this, as a PhaseOrdering test, do you have in mind a reduced version of `_ZN8zvariant19framing_offset_size17FramingOffsetSize12write_offset17h7af96b41ab784441E` (bench/zed-rs/original/8bnapxt4ilkd5y3egr7fzm1sv.ll, where the regression appears), and test it with default O2, correct?

Yes

antoniofrighetto committed:
[PhaseOrdering] Introduce test for PR145477 (NFC)

antoniofrighetto committed:
[SimplifyCFG] Do not run `simplifySwitchOfPowersOfTwo` in early invocations

It may be desirable not to carry out `simplifySwitchOfPowersOfTwo`
transform during early pipeline invocations, so as to let other
optimizations occur first.

antoniofrighetto commented:
Added, thanks (not exactly testing the original regression, but still managed to show an improvement w/ ConvertSwitchToLookupTable on).
